torch/export/unflatten.py

```diff
@@ -1026,32 +1048,47 @@ def _sink_params(
     # Also remove from call_module nodes
     call_module_nodes = filter(lambda n: n.op == "call_module", graph.nodes)
     for node in call_module_nodes:
-        node.args = tuple(filter(lambda n: n.name not in inputs_to_state, node.args))
+        submodule = _recursive_getattr(module, node.target.split("."))
+        if submodule is not None and id(submodule) in module_id_to_inputs_removed:
+            node.args = tuple(
+                filter(
+                    lambda n: n.name not in module_id_to_inputs_removed[id(submodule)],
+                    node.args,
+                )
+            )
 
+    # Filter out inputs_to_state corresponding to current scope.
+    inputs_to_state_of_scope: Dict[torch.fx.Node, list[str]] = {}
```

Comment:
Hmm could you actually remind me why we need to split this up into 2 loops now? I don't actually remember...

Reply:
The first loop: `for node in inputs` filters the `inputs_to_state` such that the inputs node **of current scope** is put in `inputs_to_state_of_scope`.
The second loop: `for node, state_name in inputs_to_state_of_scope.items()` actually operate on those nodes.
And yes, I agree that the two loops can be merged too.